### PR TITLE
Add support for triggering change handlers when using :open-editor

### LINKED ISSUE #3786: Changing text with external editor doesn't run JS handlers
When e.g. doing `<Ctrl-e>` in a GitLab issue and adding some text, the comment button will stay grayed out.

### PR DESCRIPTION
**Pull request by jgkamat**
Closes #3786

`input` was the most logical change handler to trigger when we change text, as we don't need to provide any additional information to fire it. There are a couple of other good options though (such as `paste`) but they seem less general.

I don't know of any other websites that use this behavior other than gitlab, if someone could come up with a list, I could test them out and change the signal we emit if there are differences.

I'm not sure if it would be worth it to add tests for this (it would be a little bit of extra work to set up a page that checked for handlers firing), let me know if we do want them though.

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/4204)

### CODE REVIEW

File: qutebrowser/javascript/webelem.js [hunk 1]
```diff
@@ -362,6 +362,11 @@ window._qutebrowser.webelem = (function() {
         document.execCommand("insertText", false, text);
     };
 
+    funcs.dispatch_event = (id, event) => {
+        const elem = elements[id];
+        elem.dispatchEvent(new Event(event));
```
Comment (The-Compiler) [inline qutebrowser/javascript/webelem.js]: The [MDN docs](https://developer.mozilla.org/en-US/docs/Web/Events/input) for `InputEvent` say that it bubbles up to other elements, while the [docs](https://developer.mozilla.org/en-US/docs/Web/API/Event/Event) for the `Event()` constructor says that `bubbles` is `false` by default.

I guess we should add `bubbles`, `cancleable` and `composes` (with the proper default values) to the Python side, pass them to JS as well, and pass them to `new Event`.

Comment (jgkamat) [reply qutebrowser/javascript/webelem.js]: This is a little odd, when running `new InputEvent("input")` `bubbles` is false. Maybe the `bubbles` property is set to true by default when the event is emitted from the browser (so I'll try doing that here as well).

File: qutebrowser/browser/webkit/webkitelem.py [hunk 2]
```diff
@@ -128,13 +128,16 @@ def set_value(self, value):
...
-            raise webelem.OrphanedError("Tab containing element vanished")
         log.webelem.debug("Firing event on {!r} via javascript.".format(self))
-        self._elem.evaluateJavaScript("this.dispatchEvent(new Event('{}'))"
-                                      .format(event))
+        event = javascript.string_escape(event)
+        self._elem.evaluateJavaScript(
+            "this.dispatchEvent(new Event('{}', "
+            "{{'bubbles': {}, 'cancelable': {}, 'composed': {}}}))"
+            .format(event, str(bubbles).lower(), str(cancelable).lower(),
+                    str(composed).lower()))
```
Comment (The-Compiler) [inline qutebrowser/browser/webkit/webkitelem.py]: You should probably use `javascript.convert_js_arg()` for all args (including `event`) here instead (feel free to make it public)

File: qutebrowser/javascript/webelem.js [hunk 3]
```diff
@@ -362,6 +362,16 @@ window._qutebrowser.webelem = (function() {
         document.execCommand("insertText", false, text);
     };
 
+
```
Comment (toofar) [inline qutebrowser/javascript/webelem.js]: That's an extra blank line.

**Pull request merged**